app/src/routes/Tournaments/components/Match.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { PureComponent } from "react";
+import PropTypes from "prop-types";
+import "./Match.scss";
+import { List, ListItem, ListItemAction, ListItemContent } from "react-mdl/lib/List";
+import { Checkbox } from "react-mdl/lib";
+
+const propTypes = {
+    host: PropTypes.object.isRequired,
+    guest: PropTypes.object.isRequired,
+    winner: PropTypes.string.isRequired,
+    setWinner: PropTypes.func.isRequired,
+    updateNextRoundButton: PropTypes.func.isRequired
+};
+
+const defaultProps = {};
+
+class Match extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.winCallback = this.winCallback.bind(this);
+        this.winNokCallback = this.winNokCallback.bind(this);
+        this.win = this.win.bind(this);
+        this.setCheckboxes = this.setCheckboxes.bind(this);
+        this.setCheckboxes(this.props, null);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.setCheckboxes(nextProps, this.props);
+    }
+
+    setCheckboxes(propsNew, propsOld) {
+        if (propsOld == null || propsNew !== propsOld) {
+            this.state = {
+                radiosDisabled: propsNew.winner != "0",
+                hostCheckboxValue: propsNew.winner == "1" ? 1 : 0,
+                guestCheckboxValue: propsNew.winner == "-1" ? 1 : 0
+            };
+        }
+    }
+
+    win = (winner) => {
+        this.props.setWinner(winner, () => this.winCallback(winner === this.props.host), this.winNokCallback);
+    };
+
+    winCallback(hostWon) {
+        this.setState({
+            hostCheckboxValue: hostWon ? 1 : 0,
+            guestCheckboxValue: hostWon ? 0 : 1,
+            radiosDisabled: true
+        });
+        this.props.updateNextRoundButton();
+    }
+
+    winNokCallback() {
+        this.setState({
+            hostCheckboxValue: 0,
+            guestCheckboxValue: 0
+        });
+    }
+
+    render() {
+        return (
+            <div>
+                <List style={{ width: "300px" }}>
+                    <ListItem>
```

Comment:
lot of repeating content which could be replaced with `[this.props.host, this.props.guest].map(...`